### PR TITLE
Add style for xkcd

### PR DESCRIPTION
**Pull request by souravsingh**
Added a style file for xkcd
Fixes https://github.com/matplotlib/matplotlib/issues/5992

### CODE REVIEW

File: lib/matplotlib/mpl-data/stylelib/xkcd.mplstyle [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+##AXES
+
+axes.linewidth     : 1.5
+axes.grid          : False
+axes.unicode_minus : False
+axes.edgecolor     : black
+
+##PATH
+
+path.sketch        : (scale, length, randomness)
```
Comment (mdboom) [inline lib/matplotlib/mpl-data/stylelib/xkcd.mplstyle]: Shouldn't these be numbers, rather than the placeholders "scale, length, randomness"?

**Comment by mdboom** [comment]
This is missing the patheffect that puts a white border behind all lines.

**Comment by afvincent** [comment]
Ping @souravsingh : are you still interested working on this style sheet? With @mdboom's remarks, I guess one should get something that starts to resemble xkcd plots :).

**Comment by souravsingh** [comment]
@afvincent I am interested in working on the style sheet. I am confused about the parameter for patheffect

**Comment by WeatherGod** [comment]
The patheffect part is probably going to be the hardest part at the moment,
since it doesn't exist as a style parameter yet.

**Comment by afvincent** [comment]
@WeatherGod You're right: the patheffect part might be a bit more difficult to setup than I expected :/. I was fooled by the example of `rcParams['path.effects'] = [patheffects.withStroke(linewidth=4, foreground="w")]` where the relevant rcParams key exists. But `patheffects` is actually loaded elsewhere before…

**Comment by afvincent** [comment]
@souravsingh Apparently, there is an official xkcd font, so it could be nice to add it as the default choice (another PR is open to add it to `plt.xkcd`, see #6811 for details). I don't have a solution for the `patheffect` part yet, but I'll try to have a look at this problem during the weekend.

**Comment by souravsingh** [comment]
@afvincent Is the problem fixed?

**Comment by afvincent** [comment]
Wohh I totally forgot this one : apologies @souravsingh! The xkcd font seems to have been added to the list so this part should be OK. 

Unfortunately, I cannot see an easy solution to the `patheffect` part 😞. The validator of the corresponding rcparam (`rcParams['path.effects']`) would require to `eval` a string like "patheffects.withStroke(linewidth=4, foreground='w')", which may raise security issues if I am correct.  *But maybe a core dev. will prove me wrong ^^.*

**Comment by dopplershift** [comment]
@afvincent Well, we control that string, and `patheffects`, so it's certainly possible to create some kind of string syntax that we can then parse ourselves without using `eval`....

**Comment by jklymak** [comment]
If we parsed `path.effect` as a string like "withStroke(linewidth=4, foreground="w")" and then did something like:

```python
patheffect = getattr(patheffects.modules[__name__], str)
rc['path.effect'] = patheffect(**kwargs)
```

would that be safe enough?    ping @anntzer

**Comment by anntzer** [comment]
*Please* no more parsing craziness.

**Comment by souravsingh** [comment]
Is there a resolution for the PR?

**Comment by jklymak** [comment]
I'll close in favour of #14943

**Pull request closed without merging**